### ISSUE TITLE
Can't create and start container

### ISSUE THREAD

**Issue by feiskyer** [hyperhq/hyperd#544]
Create container based on image `gcr.io/google_containers/k8s-dns-sidecar-amd64` failed, and no logs are found for the created container:

```
# hyperctl run -d gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.1
POD id is k8s-dns-sidecar-amd64-1.14.1-1105431549
Time to run a POD is 5406 ms

# hyperctl list container | grep k8s-dns-sidecar-amd64-1.14.1-1105431549
9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448   k8s-dns-sidecar-amd64-1.14.1-1105431549                                                             k8s-dns-sidecar-amd64-1.14.1-1105431549

# hyperctl list | grep k8s-dns-sidecar-amd64-1.14.1-1105431549
k8s-dns-sidecar-amd64-1.14.1-1105431549                                                         k8s-dns-sidecar-amd64-1.14.1-1105431549                    
```

After starting the container again, it is shown as running, but the container process has actually down:

```
# hyperctl exec 9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448 ps
PID   USER     TIME   COMMAND
    1 root       0:00 /init
    4 root       0:00 ps
# hyperctl exec 9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448 ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 /init
    5 root       0:00 ps aux

# hyperctl list container | grep 9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448
9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448   k8s-dns-sidecar-amd64-1.14.1-1105431549                                                             k8s-dns-sidecar-amd64-1.14.1-1105431549                                                         running
```

Related hyperd log:  [logs.txt](https://github.com/hyperhq/hyperd/files/843762/logs.txt)

cc @laijs @gnawux

**Comment by Crazykev** [comment]
find such log fragment, seems the reason why container start failed.
```
I0315 03:14:30.698421   13728 vm_console.go:46] SB[vm-uSONjYYWGm] [CNL] try to find the user: nobody:nobody
I0315 03:14:30.707577   13728 vm_console.go:46] SB[vm-uSONjYYWGm] [CNL] can't find the user: No such file or directory
I0315 03:14:30.707791   13728 vm_console.go:46] SB[vm-uSONjYYWGm] [CNL] setup exec user failed
```

**Comment by feiskyer** [comment]
Yep, also found these logs, but `nobody` exists in `/etc/passwd`:

```
hyperctl exec 9e635efb753f476a2bb3fa4e3d5d9b41e24c059f8e4d1b43ee78a23472882448 cat /etc/passwd | grep nobody
nobody:x:65534:65534:nobody:/:/sbin/nologin
```

**Comment by gnawux** [comment]
I think this is an hyperstart issue, could @bergwolf take a look?

**Comment by bergwolf** [comment]
hyperstart is asked to find user nobody:nobody
```
try to find the user: nobody:nobody
```
however it should be user `nobody` and group `nobody`.

spec passed to hyperd is
```
"process":{"id":"init","user":"nobody:nobody","terminal":false,"stdio":1,"stderr":2,"args":["/sidecar"],"envs":[{"env":"PATH","value":"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"}],"workdir":"/"}
```

it should contain `"user":"nobody","group":"nobody"` instead.

**Comment by gnawux** [comment]
I checked this image again, which specified `"user": "nobody:nobody"`. is this correct? If so, we need handle this.... I really hate a image specified like this.

```
    {
        "Id": "sha256:fc5e302d8309a8deb3315452b428969aed4aa931189b10a3231ce48fc0687224",
        "RepoTags": [
            "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.1"
        ],
        "RepoDigests": [],
        "Parent": "",
        "Comment": "",
        "Created": "2017-02-27T18:56:29.617416621Z",
        "Container": "ec1d14a8bbb5f23b3062e513c5f5403b6525bed842db4a680388e65c40ff2b52",
        "ContainerConfig": {
            "Hostname": "11fbdc1f630f",
            "Domainname": "",
            "User": "nobody:nobody",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "ENTRYPOINT [\"/sidecar\"]"
            ],
            "Image": "sha256:2de46c332e8f9574a87bc5a5c6dfebbc0c88829e8ab8ba1757efdd4b00592c74",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/sidecar"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "DockerVersion": "1.12.6",
        "Author": "Bowei Du \u003cbowei@google.com\u003e",
        "Config": {
            "Hostname": "11fbdc1f630f",
            "Domainname": "",
            "User": "nobody:nobody",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": null,
            "Image": "sha256:2de46c332e8f9574a87bc5a5c6dfebbc0c88829e8ab8ba1757efdd4b00592c74",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/sidecar"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 44517061,
        "VirtualSize": 44517061,
        "GraphDriver": {
            "Name": "rbd",
            "Data": {
                "DeviceId": "0",
                "DeviceName": "docker_image_c299af88270190387755d7ea0ad39a704dfa72bcbeec30989fb7bead560695ab",
                "DeviceSize": "10737418240"
            }
        },
        "RootFS": {
            "Type": ""
        }
    }
```

**Comment by gnawux** [comment]
Is there anyone willing to fix this? If nobody self-assigned tonight, I will try to make a PR.

**Comment by bergwolf** [comment]
@gnawux I can handle this one. docker config user field supports:
`--user=[ user | user:group | uid | uid:gid | user:gid | uid:group ]`

**Comment by gnawux** [comment]
need regression test for this issue

**Comment by bergwolf** [comment]
I'll add regression for both when fixing https://github.com/hyperhq/hyperd/issues/548

**Comment by gnawux** [comment]
Thank you @bergwolf

**Comment by feiskyer** [comment]
Cool!

**Comment by gnawux** [comment]
fixed by #550
